test/run_envoy_bazel_coverage.sh

```diff
@@ -31,97 +21,43 @@ else
   COVERAGE_TARGETS=//test/...
 fi
 
-# This is where we are going to copy the .gcno files into.
-GCNO_ROOT=bazel-out/k8-dbg/bin/test/coverage/coverage_tests.runfiles/"${WORKSPACE}"
-echo "    GCNO_ROOT=${GCNO_ROOT}"
-
-echo "Cleaning .gcno from previous coverage runs..."
-NUM_PREVIOUS_GCNO_FILES=0
-for f in $(find -L "${GCNO_ROOT}" -name "*.gcno")
-do
-  rm -f "${f}"
-  let NUM_PREVIOUS_GCNO_FILES=NUM_PREVIOUS_GCNO_FILES+1
-done
-echo "Cleanup completed. ${NUM_PREVIOUS_GCNO_FILES} files deleted."
-
 # Make sure //test/coverage:coverage_tests is up-to-date.
 SCRIPT_DIR="$(realpath "$(dirname "$0")")"
-(BAZEL_BIN="${BAZEL_COVERAGE}" "${SCRIPT_DIR}"/coverage/gen_build.sh ${COVERAGE_TARGETS})
-
-echo "Cleaning .gcda/.gcov from previous coverage runs..."
-NUM_PREVIOUS_GCOV_FILES=0
-for f in $(find -L "${GCOVR_DIR}" -name "*.gcda" -o -name "*.gcov")
-do
-  rm -f "${f}"
-  let NUM_PREVIOUS_GCOV_FILES=NUM_PREVIOUS_GCOV_FILES+1
-done
-echo "Cleanup completed. ${NUM_PREVIOUS_GCOV_FILES} files deleted."
+NO_GCOV=1 "${SCRIPT_DIR}"/coverage/gen_build.sh ${COVERAGE_TARGETS}
```

Comment:
Maybe just remove this conditioning from `gen_build.sh`

Reply:
sure, removed as well as gcc_only_test.